src/app/components/taxonomy/TreeView.jsx

```diff
@@ -34,9 +35,61 @@ const ItemButton = props => {
   );
 };
 
+const AnimateChildNodes = props => {
+  const { node, show } = props;
+  const duration = parseInt(
+    getComputedStyle(document.body).getPropertyValue(
+      "--tree-animation-duration"
+    )
+  );
+
+  const itemHeight = 47.6;
```

Comment:
[**Discuss**] Irgendwie gefällt mir die Animationslogik nicht so richtig. Wir haben hier hardcoded Pixel-Höhen von items die genau dem CSS entsprechen muss, sonst geht alles kaputt.
Zudem starten hier JS-Timer die am Besten genau mit den CSS transition Timern übereinstimmen, sonst könnte es komisch aussehen.
Ja, das funktioniert aktuell wie es sollte, aber ich habe da ein schlechtes Bauchgefühl bei sowas :)
Ich bin generell ein Freund davon, dass bei Animation so viel wie möglich CSS überlassen wird und JS nur so wenig wie möglich macht. D.h. JS triggert nur die Animation, indem eine Klasse gesetzt/entfernt wird.
Das passiert hier ja, nur wird gleichzeitig am Ende der Animation wieder etwas geändert an den Klassen, was den JS-Timer benötigt. Und zudem wird eben hart ein inline-style gesetzt für die Höhe.
Um da evtl. eine "schönere" Lösung zu bekommen müsste ich mich mehr mit dem Code beschäftigen und herumspielen. Evtl. sollten wir das einfach nochmal im Team diskutieren, warum es evtl. diese Lösung braucht, oder ob man das etwas "schöner" hinbekommt.

Reply:
Das ist ein wenig ein Problem der aufeinander aufsetzenden PRs -- in einem späteren wird das mit dem CSS gesynct